compiler/optimizer/LocalOpts.cpp

```diff
@@ -8166,8 +8166,15 @@ TR_ColdBlockMarker::identifyColdBlocks()
    for (TR::AllBlockIterator iter(optimizer()->getMethodSymbol()->getFlowGraph(), comp()); iter.currentBlock(); ++iter)
       {
       TR::Block *block = iter.currentBlock();
+
       if (block->isCold())
+         {
+         // OSR blocks may not have had a chance to set their frequencies yet
+         if (block->isOSRCodeBlock() || block->isOSRCatchBlock())
+            block->setFrequency(0);
```

Comment:
The only cold block frequency I see being set during IlGen is `UNKNOWN_COLD_BLOCK_COUNT`. `CATCH_COLD_BLOCK_COUNT` may be another choice, but does not seem suitable for the OSRCodeBlock.